go/multipooler/manager/manager.go

```diff
@@ -1044,13 +1006,290 @@ func (pm *MultiPoolerManager) ConfigureSynchronousReplication(ctx context.Contex
+
+	// Query synchronous_standby_names
+	var syncStandbyNamesStr string
+	err := pm.db.QueryRowContext(ctx, "SHOW synchronous_standby_names").Scan(&syncStandbyNamesStr)
+	if err != nil {
+		return nil, mterrors.Wrap(err, "failed to query synchronous_standby_names")
+	}
+
+	// Only parse standby names if not empty
+	syncStandbyNamesStr = strings.TrimSpace(syncStandbyNamesStr)
+	if syncStandbyNamesStr != "" {
+		syncConfig, err := parseSynchronousStandbyNames(syncStandbyNamesStr)
+		if err != nil {
+			return nil, err
+		}
+		config.SynchronousMethod = syncConfig.Method
+		config.NumSync = syncConfig.NumSync
+		config.StandbyIds = syncConfig.StandbyIDs
+	}
+
+	// Query synchronous_commit
+	var syncCommitStr string
+	err = pm.db.QueryRowContext(ctx, "SHOW synchronous_commit").Scan(&syncCommitStr)
+	if err != nil {
+		return nil, mterrors.Wrap(err, "failed to query synchronous_commit")
+	}
+
+	// Map string to enum
+	var syncCommitLevel multipoolermanagerdata.SynchronousCommitLevel
+	switch strings.ToLower(syncCommitStr) {
+	case "off":
+		syncCommitLevel = multipoolermanagerdata.SynchronousCommitLevel_SYNCHRONOUS_COMMIT_OFF
+	case "local":
+		syncCommitLevel = multipoolermanagerdata.SynchronousCommitLevel_SYNCHRONOUS_COMMIT_LOCAL
+	case "remote_write":
+		syncCommitLevel = multipoolermanagerdata.SynchronousCommitLevel_SYNCHRONOUS_COMMIT_REMOTE_WRITE
+	case "on":
+		syncCommitLevel = multipoolermanagerdata.SynchronousCommitLevel_SYNCHRONOUS_COMMIT_ON
+	case "remote_apply":
+		syncCommitLevel = multipoolermanagerdata.SynchronousCommitLevel_SYNCHRONOUS_COMMIT_REMOTE_APPLY
+	default:
+		return nil, mterrors.New(mtrpcpb.Code_INVALID_ARGUMENT,
+			fmt.Sprintf("unknown synchronous_commit value: %q", syncCommitStr))
+	}
+	config.SynchronousCommit = syncCommitLevel
+
+	return config, nil
+}
+
 // PrimaryStatus gets the status of the leader server
-func (pm *MultiPoolerManager) PrimaryStatus(ctx context.Context) (map[string]interface{}, error) {
+func (pm *MultiPoolerManager) PrimaryStatus(ctx context.Context) (*multipoolermanagerdata.PrimaryStatus, error) {
 	if err := pm.checkReady(); err != nil {
 		return nil, err
 	}
+
 	pm.logger.Info("PrimaryStatus called")
-	return nil, mterrors.New(mtrpcpb.Code_UNIMPLEMENTED, "method PrimaryStatus not implemented")
+
+	// Check PRIMARY guardrails (pooler type and non-recovery mode)
+	if err := pm.checkPrimaryGuardrails(ctx); err != nil {
+		return nil, err
+	}
+
+	status := &multipoolermanagerdata.PrimaryStatus{}
+
+	// Get current LSN and recovery status in a single query
+	var lsn string
+	var isInRecovery bool
+	err := pm.db.QueryRowContext(ctx, "SELECT pg_current_wal_lsn()::text, pg_is_in_recovery()").Scan(&lsn, &isInRecovery)
+	if err != nil {
+		pm.logger.Error("Failed to query LSN and recovery status", "error", err)
+		return nil, mterrors.Wrap(err, "failed to query LSN and recovery status")
+	}
+	status.Lsn = lsn
+	status.Ready = !isInRecovery
```

Comment:
@deepthi this is what seemed sensible to me in this context. What do you think?

Reply:
if `isInRecovery` is true, we won't even get this far, right? we'll return on line 1315. So if this is really the primary, Ready can never be false.